go/service/gregor.go

```diff
@@ -1298,6 +1318,7 @@ func (g *gregorHandler) Reset() error {
 	g.Shutdown()
 	g.setFirstConnect(true)
 	g.shutdownGregorClient(context.TODO())
+	g.pushStateCh <- nil // clear any pending push state calls
```

Comment:
close on shutdown?

Reply:
I'm not sure how to get it going again given current construction.

Reply:
Nvm, made a change